compiler/haskell-ide-core/src/Development/IDE/Types/Options.hs

```diff
@@ -18,6 +18,13 @@ data IdeOptions = IdeOptions
   , optRunGhcSession :: forall a. Maybe ParsedModule -> PackageDynFlags -> Ghc a -> IO a
   -- ^ Setup a GHC session using a given package state. If a `ParsedModule` is supplied,
   -- the import path should be setup for that module.
+  , optLocateHieFile :: PackageConfig -> Module -> IO (Maybe FilePath)
+  -- ^ Locate the HIE file for the given module. The PackageConfig can be
+  -- used to lookup settings like importDirs.
+  , optLocateSrcFile :: PackageConfig -> Module -> IO (Maybe FilePath)
```

Comment:
Should these two options form a pair? If you don't call them together it's going to be wrong - using optLocateSrcFile separately won't go through FileStore so would be buggy.

Reply:
Moved to a separate type.